### DISCUSSION TITLE
播放时自动1.5倍速度

### DISCUSSION THREAD

**Discussion by YongHao2033** [the1812/Bilibili-Evolved#1639] [综合讨论 / General]
播放视频时自动1.5倍速度，当前版本v1.12.0。必须手动调节倍速。

**Comment by LonelySteve** [comment]
是想要实现播放视频时自动切换到 1.5 倍数？还是说即使切换到其他倍数，仍然会在每次重载视频页时，自动切换到 1.5 倍数？

如果是前者，打开「记忆上次播放速度」的功能试试？设置倍数之后可以跨页面记忆倍数。

如果是后者，可能是个 BUG？先关闭「记忆上次播放速度」功能，应该就恢复原本的视频倍数记忆策略了（只刷新页面和跨分 P 时保持倍数不变）。

**Comment by YongHao2033** [comment]
谢谢您能回复我，我遇到情况是后者，1.不管打开任何视频都是1.5倍速度播放；2.在视频播放期间我调整成正常速度一旦我暂停视频，再次播放后又变成了1.5倍速度播放视频。我也尝试关闭“记忆上次播放速度”也无济于事，我也关闭了其他油猴插件也没有用。

发自我的iPhone

**Comment by LonelySteve** [comment]
说实话有点神奇，没啥思路，理论上这是很难发生的，如果想要达成这个 bug，需要至少满足以下条件： 

- 即使关闭记忆倍速功能，也不会影响实例化一个 VideoSpeedController 对象，并进行初始化 
- Observer.videoChange 错误地在暂停再播放视频时触发回调 - 内部监听逻辑没能在倍速变更时成功刷新记忆的倍速值 而上面这些条件应该都不能达成。 

现在我好奇的一点是，如果刷新视频页但不自动播放，倍速是否仍然会被自动切换到 1.5 倍速？ 还有个问题是，就算真的是因为脚本的 bug 导致出现这个诡异的现象，那这个 1.5 倍速到底是从哪里来的？你可以通过在控制台（按F12打开开发者工具，切换到 Console 选项卡）执行以下代码： 

`bilibiliEvolved.setting.defaultVideoSpeed`  

来查看当前的记忆倍速，并且可以通过下面代码重置这个倍速值：

 `bilibiliEvolved.setting.defaultVideoSpeed = "1"` 

虽然这么做可能不能解决 bug，但是应该可以给我提供一些思路。

**Comment by YongHao2033** [comment]
不好意思，白天头痛休息了一天，我这边打开电脑后还是遇到昨天的问题，我这边录制了一段视频和几张截图，不知道有没有参考价值。








------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "the1812/Bilibili-Evolved"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月13日(星期六) 上午10:37
***@***.***&gt;;
抄送:&nbsp;"🎶I see ***@***.******@***.***&gt;;
主题:&nbsp;Re: [the1812/Bilibili-Evolved] 播放时自动1.5倍速度 (#1639)





 说实话有点神奇，没啥思路，理论上这是很难发生的，如果想要达成这个 bug，需要至少满足以下条件： - 即使关闭记忆倍速功能，也不会影响实例化一个 VideoSpeedController 对象，并进行初始化 - Observer.videoChange 错误地在暂停再播放视频时触发回调 - 内部监听逻辑没能在倍速变更时成功刷新记忆的倍速值 而上面这些条件应该都不能达成。 现在我好奇的一点是，如果刷新视频页但不自动播放，倍速是否仍然会被自动切换到 1.5 倍速？ 还有个问题是，就算真的是因为脚本的 bug 导致出现这个诡异的现象，那这个 1.5 倍速到底是从哪里来的？你可以通过在控制台（按F12打开开发者工具，切换到 Console 选项卡）执行以下代码： `bilibiliEvolved.setting.defaultVideoSpeed`  查看当前的记忆倍速，并且可以通过下面代码重置这个倍速值： `bilibiliEvolved.setting.defaultVideoSpeed = &amp;quot;1&amp;quot;` 虽然这么做可能不能解决 bug，但是应该可以给我提供一些思路。 ------------------&amp;nbsp;Original&amp;nbsp;------------------ 
 ***@***.***&amp;gt; 
 Date:&amp;nbsp;Sat, Mar 13, 2021 09:36 AM 
 ***@***.***&amp;gt;; 
 ***@***.******@***.***&amp;gt;; 
 Subject:&amp;nbsp;Re: [the1812/Bilibili-Evolved] 播放时自动1.5倍速度 (#1639) 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.


从QQ邮箱发来的超大附件

示例.mkv (11.63M, 无限期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?k=0d6138317ac674ce6e871cbd1331031e01585d53070703091d585d03041c0553560015500505011c53540f095400040309005b55353931fb8ea1c51f585a47310d&t=exs_ftn_download&code=0a815111

**Reply by LonelySteve** [reply]
虽然你说过即使关闭其他油猴脚本，仍然能复现这个问题，但是你有没有启用某些会控制倍速的浏览器插件呢？

因为我看视频中播放器在播放前会悬浮显示一个视频倍数，这个并不是本脚本的功能。

![image](https://user-images.githubusercontent.com/34429322/111032474-5f384600-8447-11eb-8674-cf6697c72404.png)

**Comment by YongHao2033** [comment]
我最后关闭了“HTML5增强”的脚本，是它的问题，我关掉之后就没有再出现问题了

发送自 Windows 10 版邮件应用

发件人: JLoeve
发送时间: 2021年3月13日 22:02
收件人: the1812/Bilibili-Evolved
抄送: YongHao2033; Author
主题: Re: [the1812/Bilibili-Evolved] 播放时自动1.5倍速度 (#1639)

虽然你说过即使关闭其他油猴脚本，仍然能复现这个问题，但是你有没有启用某些会控制倍速的浏览器插件呢？
因为我看视频中播放器在播放前会悬浮显示一个视频倍数，这个并不是本脚本的功能。

**Comment by YongHao2033** [comment]
谢谢您为我们带了这么好的脚本，非常感谢为爱发电，您也是第一个愿意回复我反馈bug的朋友，幸苦您了。十分感谢

发自我的iPhone

**Reply by LonelySteve** [reply]
不用那么客气哈，因为这块的功能是我写的，因此会比较在意，以后如果在使用本脚本情况下，遇到倍数相关或者夜间模式的 BUG，欢迎反馈，我有精力就会尽快修掉的。

**Comment by YongHao2033** [comment]
谢谢，很羡慕会写JS脚本，我也是正在学习Web前端，希望尽快能写出我的第一个像模像样的程序。

发自我的iPhone
